packages/kitsu-core/src/deepEqual/index.js

```diff
@@ -0,0 +1,56 @@
+/**
+ * Compare two objects equality 
+ *
+ * @param {object} firstObject First Object
+ * @param {object} secondObject Second Object to compare with the first one
+ * @returns {boolean} A boolean that indicates if objects are equal
+ * @private
+ * @example <caption>Deep equality check</caption>
+ * isDeepEqual({
+ *   firstName: 'John',
+ *   lastName: 'Doe',
+ *   age: 35
+ * },{
+ *   firstName: 'John',
+ *   lastName: 'Doe',
+ *   age: 35
+ * }) // true
+ */
+export const isDeepEqual = (object1, object2) => {
+  const objKeys1 = Object.keys(object1);
+  const objKeys2 = Object.keys(object2);
```

Comment:
Function should check both values are objects before contining to access `Object.keys` on them. 

Tests are failing due to `undefined` and `null` values being passed in through `object1` and/or `object2`.